app/celery/scheduled_tasks.py

```diff
@@ -143,15 +145,57 @@ def switch_current_sms_provider_on_slow_delivery():
                 dao_reduce_sms_provider_priority(provider_name, time_threshold=timedelta(minutes=10))
 
 
+def _check_slow_text_message_delivery_reports_and_raise_error_if_needed(reports: list[SlowProviderDeliveryReport]):
+    total_notifications = sum(report.total_notifications for report in reports)
+    slow_notifications = sum(report.slow_notifications for report in reports)
+    percent_slow_notifications = (slow_notifications / total_notifications) * 100
+
+    # If over 10% of all text messages sent over the period have taken longer than 5 minutes to deliver, let's flag a
```

Comment:
No action needed: I think this is a good starting metric and threshold. This is because 5 minutes is the same period that we use for our sms load balancing. 

We may decide to increase the threshold but we can see what happens.

https://grafana-notify.cloudapps.digital/d/icjsQ-MWk2/notify-summary?orgId=1&viewPanel=51&from=now-7d&to=now

Looking at the graph for sms delivery slow than 5 minutes over the last 7 days, a very very rough eyeball gives me about 7 points in the past 7 days where both sms providers were both above 10%. Assuming traffic was split equally then that would breach this alert.

Reply:
Yep, definitely something we'll need to keep an eye on if it's looking flappy or noisy. Possibly we need to only alert on it if there's 2 or 3 slow errors in a row (ie there's prolonged delivery issues)